src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseFeedbackCreationService.java

```diff
@@ -0,0 +1,211 @@
+            "org.awaitility.core.ConditionTimeoutException", "Timed?OutException");
+
+    private final ProfileService profileService;
+
+    public ProgrammingExerciseFeedbackCreationService(ProfileService profileService) {
+        this.profileService = profileService;
+    }
+
+    /**
+     * Filters and processes a feedback error message, thereby removing any unwanted strings depending on
+     * the programming language, or just reformatting it to only show the most important details.
+     *
+     * @param programmingLanguage The programming language for which the feedback was generated
+     * @param projectType         The project type for which the feedback was generated
+     * @param message             The raw error message in the feedback
+     * @return A filtered and better formatted error message
+     */
+    private String processResultErrorMessage(final ProgrammingLanguage programmingLanguage, final ProjectType projectType, final String message) {
+        final String timeoutDetailText = "The test case execution timed out. This indicates issues in your code such as endless loops, issues with recursion or really slow performance. Please carefully review your code to avoid such issues. In case you are absolutely sure that there are no issues like this, please contact your instructor to check the setup of the test.";
+        final String exceptionPrefix = "Exception message: ";
+        // Overwrite timeout exception messages for Junit4, Junit5 and other
+        // Defining two pattern groups, (1) the exception name and (2) the exception text
+        Pattern findTimeoutPattern = Pattern.compile("^.*(" + String.join("|", timeoutException) + "):?(.*)");
+        Matcher matcher = findTimeoutPattern.matcher(message);
+        if (matcher.find()) {
+            String exceptionText = matcher.group(2);
+            return timeoutDetailText + "\n" + exceptionPrefix + exceptionText.trim();
+        }
+        // Defining one pattern group, (1) the exception text
+        Pattern findGeneralTimeoutPattern = Pattern.compile("^.*:(.*timed out after.*)", Pattern.CASE_INSENSITIVE);
+        matcher = findGeneralTimeoutPattern.matcher(message);
+        if (matcher.find()) {
+            // overwrite Ares: TimeoutException
+            String generalTimeOutExceptionText = matcher.group(1);
+            return timeoutDetailText + "\n" + exceptionPrefix + generalTimeOutExceptionText.trim();
+        }
+
+        // Filter out unneeded Exception classnames
+        if (programmingLanguage == ProgrammingLanguage.JAVA || programmingLanguage == ProgrammingLanguage.KOTLIN) {
+            var messageWithoutStackTrace = message.lines().takeWhile(IS_NOT_STACK_TRACE_LINE).collect(Collectors.joining("\n")).trim();
+
+            // the feedback from gradle test result is duplicated on bamboo therefore it's cut in half
+            if (projectType != null && projectType.isGradle() && profileService.isBamboo()) {
```

Comment:
This line is the main change of this PR: Only cut the feedback on a bamboo setup.

Reply:
Is Bamboo it always cutting them in half or does it have some maximum length?